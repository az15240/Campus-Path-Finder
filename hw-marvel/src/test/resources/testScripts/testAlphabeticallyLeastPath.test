# Test alphabetically least paths. Only four nodes are provided but various edges are between them.

CreateGraph g

AddNode g n1
AddNode g n2
AddNode g n3
AddNode g n4

AddEdge g n1 n2 e2
AddEdge g n1 n2 e3
AddEdge g n1 n3 e1
AddEdge g n1 n3 e4
AddEdge g n2 n4 e7
AddEdge g n2 n4 e8
AddEdge g n3 n4 e5
AddEdge g n3 n4 e6

ListNodes g

ListChildren g n1

FindPath g n1 n2

FindPath g n1 n3

FindPath g n1 n4

FindPath g n2 n3

FindPath g n2 n4

FindPath g n3 n4

FindPath g n1 n1

FindPath g n2 n1