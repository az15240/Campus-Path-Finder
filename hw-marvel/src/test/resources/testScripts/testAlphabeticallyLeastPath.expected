# Test alphabetically least paths. Only four nodes are provided but various edges are between them.

created graph g

added node n1 to g
added node n2 to g
added node n3 to g
added node n4 to g

added edge e2 from n1 to n2 in g
added edge e3 from n1 to n2 in g
added edge e1 from n1 to n3 in g
added edge e4 from n1 to n3 in g
added edge e7 from n2 to n4 in g
added edge e8 from n2 to n4 in g
added edge e5 from n3 to n4 in g
added edge e6 from n3 to n4 in g

g contains: n1 n2 n3 n4

the children of n1 in g are: n2(e2) n2(e3) n3(e1) n3(e4)

path from n1 to n2:
n1 to n2 via e2

path from n1 to n3:
n1 to n3 via e1

path from n1 to n4:
n1 to n2 via e2
n2 to n4 via e7

path from n2 to n3:
no path found

path from n2 to n4:
n2 to n4 via e7

path from n3 to n4:
n3 to n4 via e5

path from n1 to n1:

path from n2 to n1:
no path found